cmd/soroban-rpc/internal/db/event.go

```diff
@@ -81,12 +96,12 @@ func (eventHandler *eventHandler) InsertEvents(lcm xdr.LedgerCloseMeta) error {
 			Columns("id", "ledger_sequence", "application_order", "contract_id", "event_type")
 
 		for index, e := range txEvents {
-			var contractId []byte
+			var contractID string
 			if e.Event.ContractId != nil {
-				contractId = e.Event.ContractId[:]
+				contractID = strkey.MustEncode(strkey.VersionByteContract, (*e.Event.ContractId)[:])
```

Comment:
Please don't store strkeys in the DB (they occupy a lot more space). Just store the 32 byte hash.